example_test.go

```diff
@@ -93,16 +128,74 @@ func ExampleQuery_GroupJoin() {
 		From(fruits),
 		func(i rune) rune { return i },
 		func(i string) rune { return []rune(i)[0] },
-		func(outer rune, inners []interface{}) interface{} {
+		func(outer rune, inners []interface{}) KeyValue {
 			return KeyValue{string(outer), inners}
-		})
+		},
+	)
 
 	fmt.Println(q.Results())
 
 	// Output:
 	// [{a [apple apricot]} {b [banana]} {c [cherry clementine]}]
 }
 
+func ExampleQuery_GroupJoin_2() {
+
+	type Person struct {
+		Name string
+	}
+
+	type Pet struct {
+		Name  string
+		Owner Person
+	}
+
+	magnus := Person{Name: "Hedlund, Magnus"}
+	terry := Person{Name: "Adams, Terry"}
+	charlotte := Person{Name: "Weiss, Charlotte"}
+
+	barley := Pet{Name: "Barley", Owner: terry}
+	boots := Pet{Name: "Boots", Owner: terry}
+	whiskers := Pet{Name: "Whiskers", Owner: charlotte}
+	daisy := Pet{Name: "Daisy", Owner: magnus}
+
+	people := []Person{magnus, terry, charlotte}
+	pets := []Pet{barley, boots, whiskers, daisy}
+
+	// Create a slice where each element is a KeyValue
+	// that contains a person's name as the key and a slice of strings
+	// of names of the pets they own as a value.
+
+	q := []KeyValue{}
+	From(people).GroupJoin(From(pets),
+		func(p Person) Person { return p },
+		func(p Pet) Person { return p.Owner },
+		func(person Person, pets []interface{}) KeyValue {
```

Comment:
So, in this method the conversion is not possible because the [innerLookup](https://github.com/cleitonmarx/go-linq/blob/188ac234473744c30addd02e7e8fd296e633de1f/groupjoin.go#L89) object is an interface slice.